### ISSUE TITLE
(Proposal) The four facets of this game and a road to balance?

### ISSUE THREAD

**Issue by PhoenixRiver** [naev/naev#2083]
I would like this to be aimed at higher level ideas that broaden the scope of areas that are not fleshed out within the game that contribute to these four areas

Destruction (Combat / Evolving galaxy):
This feels like it has a good spread from novice to end-game, sure, some of the larger ships don't yet have much use but then the galaxy hasn't evolved much yet... mostly waiting on content I suppose? But:

- [ ] Make stealth more than for just running away (mines / ambushes / backstabs)
- [ ] Greater mission development that allows players to help one faction overtake the rest (or at least seem to)
- [ ] More use of unidiff to impact (literally) changes to the galaxy

Construction (Trade / Evolving galaxy):
Lacks breadth and depth though the basics of the necesasry systems are already in place I feel the problems are more to do with depths in other areas:

- [ ] Make stealth more than for just running away (contraband - faction based)
- [ ] Can traders have an actual impact on the economy?
- [ ] If yes then the economy system needs to be made to reflect that - e.g. trading in rare commodities and limits on how much can be purchased or else massively increase traders ability to haul _huge_ amounts of material around but reduce profits on common commodities to near nothing and make mission trading about specialist commodities rather than generic ones
- [ ] if no then use that to broaden the number of commodities (we have 8(?) vs tens of weapon systems (guns, turrets, missile systems, fighters etc.) as well as increase variability of what can be traded where
- [ ] A lot more faction missions related to trade that causes unidiff or for reasons of lore that involve constructing a better trading environment - setting up new settlements in open space, construction projects needing lots of specialist parts or just raw material, either insystem, in faction or interfaction (whether "legal" or not)
- [ ] An insystem (no jump drive) ship that is just HUGE for transporting cargo from places where it is produced to places it is needed (bacuse so many systems have lore referring to this and yet there is often little or NO such traffic between these places e.g. the argiculture stations to local planets, the world being strip mined to build the Empire capital planet, the energy producing station near Sirius Core transporting energy to other places that need it. Whether the player can buy it or not probably depends on the above but insystem profits would need minute

Exploration (Lore / Environment):
There is a good degree of environmental exploration possible though it is pretty tough until the player masters combat in highly specilist ships and this makes lore exploration harder bacuse the majority of it is hidden behind combat faction missions.

- [ ] More missions that are not combat orientated that involve lore
- [ ] Expand the universe beyond what is actually heavily used by the factions cf #1866
- [ ] Resupply vessels that maybe even jump into systems that have emergency distress beacons
- [ ] Working emergency distress beacons for the player (and ai?) that actually bring aid - maybe an outfit you have to buy (so the pirate combat players don't suffer too much from every ship having one
- [ ] More interlinking between lore on planets, systems, events and factions
- [ ] Use the traders faction or expand it to be clan based like the pirates with clans either overlapping house factions or specific trade routes (study the safe lanes to find those) or split vertically along commodity lines and their primary producers (e.g. water/food clan surrounding / owning agriculture stations, ore/industrial goods clan surrounding/owning mining colonies

Socialisation (Interactive Lore / History or His/Her/Their Story {as I think of it}):
Some of the events and the missions already fulfil this, though proficiency in combat (and trading to a very minor degree) is a requirement for most faction missions and quite a lot of other interactive lore. The logging mechanism helps a good deal with the History aspect and could become rather more with some work.

- [ ] Expand logging to be used by every unique mission and interlink entries
- [ ] Expand missions (unique or semi-unique) to include a lot more non-combat, more social aspects of the galaxy (maybe even faction missions could involve more diplomacy)
- [ ] Expand what is know about the players actions in the past beyond a simple reputation number to allow mission makers to tailor missions for players interested in one of the four facets of the game e.g. track number of kills (in mission/event and out of mission/event), mission by type, faction and final status (success, failure, abandonment), interaction with the shipai, interaction with bar patrons, economic indicators (trade profit, volume; within system, faction, galaxy, commodity preferences)
- [ ] Have logging record mission tracking data so future missions / events can use it to alter future missions / events and/or just build a (possibly sometimes-forgets-over-time) reputation system for the player and maybe also NPCs the player meets repetedly (e.g. in missions)
- [ ] Have more NPCs that the player can catch up with or meet repeatedly who react based on what they know of the player, number one for testing this is the shipai who has the potential to be in constant contact with the player and also to know everything that they do (pretty much)
- [ ] Expand the setup for the shipai, work out what the player might want from it and try to tailor it to that by e.g. including/excluding it from events/missions, making it witty or robotic etc., have it provide a running commentary (maybe a box on the other side of the radar to mirror the comms feed) of what it knows and sees vs have it only provide anything at all when the player specifically invokes it (e.g. for tutorials only with an option to turn even that automatic interaction off)
- [ ] Autobuild a database of significant people the player has interacted with, who they are, faction, interactions giving the player the ability to "forget" people if they want (of course the system never forgets ;) to allow the player to remember better the lore they have been a part of and use this information in future missions, particularly if it is linked with event and mission logging.

I don't know about you but I see a slight bias in the above; the reason I feel this is there is because of an inbalance that already exists... possibly, partially, due to Naev's historical leaning towards EV but I'd argue that the early series involved the non-combat facets more if only because trading was simplistic, even early on exploration was easier, lore was prolific and most missions were done through the same few people and finally because the takeaway from the games is largely the non-combat parts of the games, not the minutae of the combat grind! (Though, clearly, the storyline destruction was also memorable!)

Anyway, I have definitely missed things but I am happy to edit this list to add broad-brushes where people think I have missed something important and to generalise things already on this list to maintain a nice broadstroke consistancy.

Please comment or even just note pull requests or issues that you think contribute to completing things or are relevant to particular points :)

**Comment by bobbens** [comment]
Some quick comments:

> Destruction (Combat / Evolving galaxy): This feels like it has a good spread from novice to end-game, sure, some of the larger ships don't yet have much use but then the galaxy hasn't evolved much yet... mostly waiting on content I suppose? But:
> 
>     * [ ]  Make stealth more than for just running away (mines / ambushes / backstabs)

There's some work in this direction, the stealth line of bioships gives a large damage bonus for a few seconds after stealth. I want to also add similar normal outfits in this route and allow "backstab" type gameplay.

>     * [ ]  Greater mission development that allows players to help one faction overtake the rest (or at least seem to)
>     * [ ]  More use of unidiff to impact (literally) changes to the galaxy

These two are planned, but it's pretty dangerous to do this in a more general way. In particular, the chapter system ( #1887 ) will allow large scale changes to the universe. Smaller stuff we can probably get away with unidiffs like we do now. However, we have to take care and probably will have to expand Lua libraries for this.

> Construction (Trade / Evolving galaxy): Lacks breadth and depth though the basics of the necesasry systems are already in place I feel the problems are more to do with depths in other areas:
>     * [ ]  Make stealth more than for just running away (contraband - faction based)

Contraband exists though? I do want to make it not necessarily mission-related. Also, I would like to have more stealth-based gameplay without having to relay necessarily on missions.

>     * [ ]  Can traders have an actual impact on the economy?

This is a tricky question. I don't really think the player can have an impact given the largeness of the universe, but it might happen with really rare stuff if the player can become the dominant source.

>     * [ ]  A lot more faction missions related to trade that causes unidiff or for reasons of lore that involve constructing a better trading environment - setting up new settlements in open space, construction projects needing lots of specialist parts or just raw material, either insystem, in faction or interfaction (whether "legal" or not)

Yeah, I would not do this with unidiffs but some sort of new mechanism or maybe just a saving events. I liked how EV did it with temporary price fluctuations and the likes. We do have to revamp a bit the economy any way though.

>     * [ ]  An insystem (no jump drive) ship that is just HUGE for transporting cargo from places where it is produced to places it is needed (bacuse so many systems have lore referring to this and yet there is often little or NO such traffic between these places e.g. the argiculture stations to local planets, the world being strip mined to build the Empire capital planet, the energy producing station near Sirius Core transporting energy to other places that need it. Whether the player can buy it or not probably depends on the above but insystem profits would need minute

Ships with no jump drives would be interesting to have.

> Exploration (Lore / Environment): There is a good degree of environmental exploration possible though it is pretty tough until the player masters combat in highly specilist ships and this makes lore exploration harder bacuse the majority of it is hidden behind combat faction missions.
>     * [ ]  More missions that are not combat orientated that involve lore

Definitely. I want to have missions that reward you for fully exploring areas of the map. Also @V-ktor wanted to do a campaign on having the player have to find out what happened to the different colony ships. Furthermore, with mini-games and the likes it should be possible to do even more things. Hell, the new mining stuff is already a significant step in this direction and just needs more mechanics to encourage the player to mine (planned, but haven't gotten around to it yet)

>     * [ ]  Expand the universe beyond what is actually heavily used by the factions cf [[Proposal] Uninhabited Frontier #1866](https://github.com/naev/naev/issues/1866)

I have been working on that. Still infinite things to do.

>     * [ ]  Resupply vessels that maybe even jump into systems that have emergency distress beacons

Sounds cool.

>     * [ ]  Working emergency distress beacons for the player (and ai?) that actually bring aid - maybe an outfit you have to buy (so the pirate combat players don't suffer too much from every ship having one

Sounds interesting.

>     * [ ]  More interlinking between lore on planets, systems, events and factions

Definitely would be nice once things settle down a bit more to start polishing it all much more. We're still doing quite big changes so I don't think it might be the best idea to do really fine interlinking stuff yet.

>     * [ ]  Use the traders faction or expand it to be clan based like the pirates with clans either overlapping house factions or specific trade routes (study the safe lanes to find those) or split vertically along commodity lines and their primary producers (e.g. water/food clan surrounding / owning agriculture stations, ore/industrial goods clan surrounding/owning mining colonies

Yes, I would like to have the trader guild reintroduced (with a better name), and actually have them be interconnected with everything and make more sense (tm).
 
> Socialisation (Interactive Lore / History or His/Her/Their Story {as I think of it}): Some of the events and the missions already fulfil this, though proficiency in combat (and trading to a very minor degree) is a requirement for most faction missions and quite a lot of other interactive lore. The logging mechanism helps a good deal with the History aspect and could become rather more with some work.
>     * [ ]  Expand logging to be used by every unique mission and interlink entries
>     * [ ]  Expand missions (unique or semi-unique) to include a lot more non-combat, more social aspects of the galaxy (maybe even faction missions could involve more diplomacy)

Sounds good.

>     * [ ]  Expand what is know about the players actions in the past beyond a simple reputation number to allow mission makers to tailor missions for players interested in one of the four facets of the game e.g. track number of kills (in mission/event and out of mission/event), mission by type, faction and final status (success, failure, abandonment), interaction with the shipai, interaction with bar patrons, economic indicators (trade profit, volume; within system, faction, galaxy, commodity preferences)

We've actually started to track lots more player stats recently, hoping that it can be put to good use in missions and stuff. Still room to add many more.

>     * [ ]  Have logging record mission tracking data so future missions / events can use it to alter future missions / events and/or just build a (possibly sometimes-forgets-over-time) reputation system for the player and maybe also NPCs the player meets repetedly (e.g. in missions)

This can probably be done with a Lua library.

>     * [ ]  Have more NPCs that the player can catch up with or meet repeatedly who react based on what they know of the player, number one for testing this is the shipai who has the potential to be in constant contact with the player and also to know everything that they do (pretty much)

This is something I want to do. You get this a bit with the minerva campaign.

>     * [ ]  Expand the setup for the shipai, work out what the player might want from it and try to tailor it to that by e.g. including/excluding it from events/missions, making it witty or robotic etc., have it provide a running commentary (maybe a box on the other side of the radar to mirror the comms feed) of what it knows and sees vs have it only provide anything at all when the player specifically invokes it (e.g. for tutorials only with an option to turn even that automatic interaction off)

Definitely something we want to do.

>     * [ ]  Autobuild a database of significant people the player has interacted with, who they are, faction, interactions giving the player the ability to "forget" people if they want (of course the system never forgets ;) to allow the player to remember better the lore they have been a part of and use this information in future missions, particularly if it is linked with event and mission logging.

This is basically #1865 .

To summarize, I agree with many of your points. Naev is still very very far from done and there's a ton of things we still have to do. Seeing a post like this is a good reminder of all the great stuff we have left to add :)

**Comment by PhoenixRiver** [comment]
I feel like #1032 work would be for more involuable impersonal information and #1865 works for both more volatile, changable personal information as well as #1032... I'd be tempted to see if some seperation might be possible (even just different colouring) since who that one NPC who backstabs you part way through is is a very different type of information from what is the Nebula...

And when I have more of a moment I'll add in more of the Combat / Trade stuff like player fleets and being a pirate but I was keen to try and note that there are a lot of combat systems already but much less for other styles of play... I feel like they currently rank in the order I listed them above, Combat being the most fleshed out with Social being the one that is only now starting to be looked at... for a long time there was just that one FLF-DVaered mission where the player had agency outside of taking a mission on at all... sorry, GTG!

**Comment by PhoenixRiver** [comment]
Thank you @bobbens - those references have now been folded in... any more for any more?

**Comment by bobbens** [comment]
Just some updates:
* With points of interest implemented, we have the possibility of adding lore-focused non-combat missions easily.
* Soromid bioships can specialize in stealth which gives them a +damage bonus for a while after breaking stealth allowing for backstabs.
